server/rest/src/main/java/org/infinispan/rest/resources/SecurityResource.java

```diff
@@ -0,0 +1,170 @@
+import java.util.concurrent.CompletionStage;
+
+import javax.security.auth.Subject;
+
+import org.infinispan.commons.dataconversion.internal.Json;
+import org.infinispan.configuration.cache.Configuration;
+import org.infinispan.configuration.global.GlobalAuthorizationConfiguration;
+import org.infinispan.configuration.global.GlobalConfiguration;
+import org.infinispan.rest.InvocationHelper;
+import org.infinispan.rest.NettyRestResponse;
+import org.infinispan.rest.cachemanager.RestCacheManager;
+import org.infinispan.rest.framework.ResourceHandler;
+import org.infinispan.rest.framework.RestRequest;
+import org.infinispan.rest.framework.RestResponse;
+import org.infinispan.rest.framework.impl.Invocations;
+import org.infinispan.security.AuditContext;
+import org.infinispan.security.AuthorizationPermission;
+import org.infinispan.security.MutablePrincipalRoleMapper;
+import org.infinispan.security.PrincipalRoleMapper;
+import org.infinispan.security.impl.Authorizer;
+import org.infinispan.security.impl.SubjectACL;
+
+import io.netty.handler.codec.http.HttpResponseStatus;
+
+/**
+ * @since 10.1
+ */
+public class SecurityResource implements ResourceHandler {
+   private final InvocationHelper invocationHelper;
+   private final String accessGrantedPath;
+   private final String accessDeniedPath;
+   private final MutablePrincipalRoleMapper principalRoleMapper;
+
+   public SecurityResource(InvocationHelper invocationHelper, String accessGrantedPath, String accessDeniedPath) {
+      this.invocationHelper = invocationHelper;
+      this.accessGrantedPath = accessGrantedPath;
+      this.accessDeniedPath = accessDeniedPath;
+      GlobalConfiguration globalConfiguration = SecurityActions.getCacheManagerConfiguration(invocationHelper.getRestCacheManager().getInstance());
+      PrincipalRoleMapper mapper = globalConfiguration.security().authorization().roleMapperConfiguration().roleMapper();
+      this.principalRoleMapper = mapper != null && mapper instanceof MutablePrincipalRoleMapper ? (MutablePrincipalRoleMapper) mapper : null;
+   }
+
+   @Override
+   public Invocations getInvocations() {
+      return new Invocations.Builder()
+            .invocation().methods(GET).path("/v2/login").withAction("config")
+               .anonymous().handleWith(this::loginConfiguration)
+            .invocation().methods(GET, POST).deprecated().path("/v2/login").withAction("login")
+               .permission(AuthorizationPermission.NONE).name("USER LOGIN").auditContext(AuditContext.SERVER)
+               .handleWith(this::login)
+            .invocation().methods(GET).deprecated().path("/v2/login")
+               .permission(AuthorizationPermission.NONE).name("USER LOGIN").auditContext(AuditContext.SERVER)
+               .handleWith(this::login)
+            .invocation().methods(GET).path("/v2/security/user/acl")
+               .handleWith(this::acl)
+            .invocation().method(GET).path("/v2/security/roles")
```

Comment:
At my understanding; if the rest api is responding here with 
`["observer","___script_manager","application","admin","deployer","___schema_manager"]`
I can create a user with admin role
`katia-admin=admin`
 
but then when I call `http://localhost:11222/rest/v2/security/roles/katia-admin`
I get empty array as response

and `http://localhost:11222/rest/v2/security/user/acl`responds with
```json 
{"global":["LIFECYCLE","READ","WRITE","EXEC","LISTEN","BULK_READ","BULK_WRITE","ADMIN","CREATE","ALL","ALL_READ","ALL_WRITE"],"caches":{"___protobuf_metadata":[],"___script_cache":[]}}
```

But when I go to the console, the protobuf schemas endpoint don't work because the caches don't have the permissions required. the schema manager role is deprecated so is this the expected behavior? 

`http://localhost:11222/rest/v2/schemas`

`ISPN000287: Unauthorized access: subject 'Subject with principal(s): [katia-admin, RolePrincipal{name='admin'}]' lacks 'BULK_READ' permission`

Reply:
I'm fixing an issue with the "implicit" roles of the schema and script caches.

Reply:
the http://localhost:11222/rest/v2/security/roles/katia-admin

always returns an array with the subject name

I would like the subject principals to be also in the acl endpoin. is this not possible ?

Reply:
the roles endpoint returns the role mapping information only as seen from the principal mapper: it cannot and will never be able  to query a user from any realm to discover all the principals and map them to roles.